### PR TITLE
Waterloggable Tiny Pumpkins

### PR DESCRIPTION
**Pull request by Virtuoel**

### CODE REVIEW

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java [hunk 1]
```diff
@@ -21,18 +26,34 @@ public TinyPumpkinBlock(Settings blockSettings) {
 	
 	@Override
 	public BlockState getPlacementState(ItemPlacementContext placementContext) {
-		return this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		final BlockState blockState = this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		if (blockState.contains(Properties.WATERLOGGED))
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: This is not needed as the property is added in `appendProperties`.

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java [hunk 2]
```diff
@@ -21,18 +26,34 @@ public TinyPumpkinBlock(Settings blockSettings) {
...
+			final FluidState fluidState = placementContext.getWorld().getFluidState(placementContext.getBlockPos());
+			return blockState.with(Properties.WATERLOGGED, fluidState.getFluid() == Fluids.WATER);
+		}
+		
+		return blockState;
 	}
 	
+	@Override
+	public FluidState getFluidState(BlockState blockState) {
+		return blockState.contains(Properties.WATERLOGGED) && blockState.get(Properties.WATERLOGGED) ? Fluids.WATER.getStill(false) : super.getFluidState(blockState);
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java:43]: Same thing about the `contains` here.

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java [hunk 3]
```diff
@@ -21,18 +26,34 @@ public TinyPumpkinBlock(Settings blockSettings) {
 	
 	@Override
 	public BlockState getPlacementState(ItemPlacementContext placementContext) {
-		return this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		final BlockState blockState = this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
+		if (blockState.contains(Properties.WATERLOGGED))
+		{
```
Comment (vini2003) [inline src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: Fix your brackets!
`if (true) {
}`

**Review by vini2003 — CHANGES_REQUESTED** [review]

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java [hunk 4]
```diff
@@ -21,18 +26,34 @@ public TinyPumpkinBlock(Settings blockSettings) {
...
 	
+	@Override
+	public FluidState getFluidState(BlockState blockState) {
+		return blockState.contains(Properties.WATERLOGGED) && blockState.get(Properties.WATERLOGGED) ? Fluids.WATER.getStill(false) : super.getFluidState(blockState);
+	}
 	
 	@Override
-	public VoxelShape getOutlineShape(BlockState blockState, BlockView blockView, BlockPos blockPosition, EntityContext entityContext) {
-		return Y_SHAPE;
+	public BlockState getStateForNeighborUpdate(BlockState blockState, Direction direction, BlockState blockStateOther, IWorld world, BlockPos blockPosition, BlockPos blockPositionOther) {
```
Comment (Juuxel) [inline src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: The parameters are a bit long, some suggestions based on stuff seen in Yarn:
- The block state parameters could be just `state` and `otherState`.
- Similar thing for positions. Just `pos` and `otherPos`.

**Comment by Shrimpstress** [comment]
Waiting on requested changes to approve.

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java (hunk 1 above)
Comment (Shrimpstress) [reply src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: There is no reason to check if this blockState contains Properties.WATERLOGGED. It always will.

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java (hunk 2 above)
Comment (Shrimpstress) [reply src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java:43]: Same thing again.

**Review by Shrimpstress — APPROVED** [review]
Things look just fine to me now.

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java (hunk 1 above)
Comment (Virtuoel) [reply src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: Waterlogging's now configurable, so checking is important.

**Review by Juuxel — APPROVED** [review]

**Review by Draylar — APPROVED** [review]

**Review by vini2003 — APPROVED** [review]

**Pull request merged**